pkg/server/cluster_definition_handler.go

```diff
@@ -0,0 +1,95 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+)
+
+const authTokenRequestURL = "https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/token"
+const pullSecretRequestURL = "https://api.openshift.com/api/accounts_mgmt/v1/access_token"
+
+type ClusterDefinition struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+	PullSecret  string `json:"pullSecret"`
+}
+
+type DeveloperTokenResponse struct {
+	ExpiresIn        int    `json:"expires_in"`
+	RefreshExpiresIn int    `json:"refresh_expires_in"`
+	RefreshToken     string `json:"refresh_token"`
+	TokenType        string `json:"token_type"`
+	NotBeforePolicy  int    `json:"not-before-policy"`
+	SessionState     string `json:"session_state"`
+	AccessToken      string `json:"access_token"`
+}
+
+func fetchAuthToken(Username string, Password string) (tokenData *DeveloperTokenResponse, err error) {
+	resp, err := http.PostForm(authTokenRequestURL, url.Values{
+		"grant_type": {"password"},
+		"client_id":  {"uhc"},
+		"username":   {Username},
+		"password":   {Password},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&tokenData)
```

Comment:
Before reading the data, you need to check the status code. The above error check only tells you if the HTTP request succeeded. It may have been successful, but the response could have been 401.

```go
if resp.StatusCode != 200 {
    return nil, errors.New("Unauthorized")
}
```

Reply:
Done, it will need to get enhanced so the error returned includes status code etc. rather than just a string.